scheduler/src/storage.rs

```diff
@@ -0,0 +1,120 @@
+use std::collections::HashMap;
+
+/// Defining a trait called IStorage that takes a generic type T.
+trait IStorage<T> {
+    fn get(&self, id: &str) -> Option<&T>;
+    fn get_mut(&mut self, id: &str) -> Option<&mut T>;
```

Comment:
I dont understand why we need this method. Can you explain me ?

Reply:
This will return a mutable reference of the object. We will be able to modify the data.